### PR TITLE
[29_3] Modify the way users choose whether to embed tm into pdf

### PR DESCRIPTION
**Pull request by tangdouer1005**
Before the modification, the option in the preference was used to control whether the exported PDF should be embedded with tm document.
the disadvantage of this method is that it is inconvenient for users.because one would like to leave to the user to choose either to embed or not for each document. 
So I deleted the option in preference and added iterm to menu to control whether to embed or not.

### CODE REVIEW

File: src/Edit/editor.hpp [hunk 1]
```diff
@@ -597,6 +597,9 @@ class editor_rep: public simple_widget_rep {
   virtual void print_doc (url ps_name, bool to_file, int first, int last) = 0;
   virtual void print_to_file (url ps_name,
 			      string first="1", string last="1000000") = 0;
+#ifdef USE_PLUGIN_PDF
+  virtual bool attach_doc_to_exported_pdf (url pdf_name) = 0;
```
Comment (darcy-shen) [inline src/Edit/editor.hpp:601]: please use scheme to implement it

File: TeXmacs/progs/texmacs/texmacs/tm-print.scm [hunk 2]
```diff
@@ -101,6 +101,29 @@
         (buffer-close buf))
       (print-to-file fname)))
 
+(tm-define (wrapped-print-to-pdf-embeded-with-tm fname)
```
Comment (darcy-shen) [inline TeXmacs/progs/texmacs/texmacs/tm-print.scm:104]: use `use-plugin-pdf?` to see if pdf plugin is supported

File: src/Edit/editor.hpp (hunk 1 above)
Comment (tangdouer1005) [reply src/Edit/editor.hpp:601]: ok

**Review by mgubi — CHANGES_REQUESTED** [review]

**Review by mgubi — APPROVED** [review]

File: src/Texmacs/Data/new_buffer.cpp [hunk 3]
```diff
@@ -545,6 +545,13 @@ buffer_export (url name, url dest, string fm) {
   if (fm == "postscript" || fm == "pdf") {
     int old_stamp= last_modified (dest);
     vw->ed->print_to_file (dest);
+    // if the saving fime is a pdf, embed the tm into pdf
```
Comment (darcy-shen) [inline src/Texmacs/Data/new_buffer.cpp:548]: typo: fime -> format

File: TeXmacs/progs/texmacs/texmacs/tm-print.scm [hunk 4]
```diff
@@ -101,6 +101,39 @@
...
+    (if (screens-buffer?)
+      (let* ((cur (current-buffer))
+             (buf (buffer-new)))
+        (buffer-copy cur buf)
+        (buffer-set-master buf cur)
+        (switch-to-buffer buf)
+        (set-drd cur)
+        (dynamic-make-slides)
+        (print-to-file fname)
+        (unless(attach-doc-to-exported-pdf fname)
```
Comment (darcy-shen) [inline TeXmacs/progs/texmacs/texmacs/tm-print.scm:116]: put a space after unless

File: TeXmacs/progs/texmacs/texmacs/tm-print.scm [hunk 5]
```diff
@@ -101,6 +101,39 @@
...
+          (notify-now "fail to attach tm to pdf"))
+        (switch-to-buffer cur)
+        (buffer-close buf))
+      (begin
+      (print-to-file fname)
+      (unless(attach-doc-to-exported-pdf fname)
```
Comment (darcy-shen) [inline TeXmacs/progs/texmacs/texmacs/tm-print.scm:122]: put a space after unless

File: TeXmacs/progs/texmacs/texmacs/tm-print.scm [hunk 6]
```diff
@@ -101,6 +101,39 @@
...
+          (notify-now "fail to attach tm to pdf")))))
```
Comment (darcy-shen) [inline TeXmacs/progs/texmacs/texmacs/tm-print.scm:123]: fail -> Fail

An error message starting with upcased letter would be better

**Pull request merged**